pkgs/development/libraries/mdds/default.nix

```diff
@@ -5,7 +5,7 @@ stdenv.mkDerivation rec {
   version = "2.0.2";
 
   src = fetchurl {
-    url = "https://kohei.us/files/${pname}/src/${pname}-${version}.tar.bz2";
+    url = "https://gitlab.com/${pname}/${pname}/-/archive/${version}/mdds-${version}.tar.bz2";
```

Comment:
`fetchFromGitLab` might be preferred here

Reply:
`fetchFromGitLab` appears to be for pulling a specific revision of a repo. Do we not want to use the official release tarballs when they exist?

Reply:
I am happy to replace with `fetchFromGitLab` if that is standard practice, even for release assets.

Reply:
try this out, the tarballs seems to be identical
```nix
fetchFromGitLab {
  owner = "mdds";
  repo = pname;
  rev = version;
  sha256 = "sha256-jCzF0REocpnP56LfY42zlGTXyKyz4GPovDshhrh4jyo=";
}
```

Reply:
They're not. The tarball from `fetchFromGitLab` lacks all the generated autotooling and requires `nativeBuildInputs = [ autoreconfHook ];`. I think we should stick with upstream's release archives unless we have some pressing reason to go to the repo at a specific ref.

Reply:
Looks like that's because you didn't update the hash and nix used the cached version of the release tarball instead of downloading the gitlab archive.

I removed the hash and got a new hash of `sha256-ClttiGc3RmDpQWwztMtH4Vr1zo50KRyv+KT1aYNOESw=`, which gave me an identical directory compared to the `fetchFromGitHub` version. The new tarball shows up as `Source code` on gitlab too: https://gitlab.com/mdds/mdds/-/releases/2.0.2

Reply:
I do not believe that to be the case, see log below. Note the `jCzF0RE` hash which matches the one in your snippet. If it had used a cached release tarball, I would've expected to never have seen a build error which necessitated an `autoreconfHook`:

```
$ cat pkgs/development/libraries/mdds/default.nix
{ lib, stdenv, fetchFromGitLab, boost, llvmPackages }:

stdenv.mkDerivation rec {
  pname = "mdds";
  version = "2.0.2";

  src = fetchFromGitLab {
    owner = "mdds";
    repo = pname;
    rev = version;
    sha256 = "sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=";
  };

  postInstall = ''
    mkdir -p "$out/lib/pkgconfig"
    cp "$out/share/pkgconfig/"* "$out/lib/pkgconfig"
  '';

  buildInputs = lib.optionals stdenv.cc.isClang [ llvmPackages.openmp ];

  checkInputs = [ boost ];

  meta = with lib; {
    homepage = "https://gitlab.com/mdds/mdds";
    description = "A collection of multi-dimensional data structure and indexing algorithm";
    platforms = platforms.all;
    license = licenses.mit;
  };
}

$ nix build --no-link --option substituters https://cache.nixos.org -L .#mdds
warning: Git tree '/home/endgame/code/nixpkgs' is dirty
source> trying https://gitlab.com/api/v4/projects/mdds%2Fmdds/repository/archive.tar.gz?sha=2.0.2
source>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
source>                                  Dload  Upload   Total   Spent    Left  Speed
source> 100  759k    0  759k    0     0   338k      0 --:--:--  0:00:02 --:--:--  338k
source> unpacking source archive /build/archive.tar.gz?sha=2.0.2
error: hash mismatch in fixed-output derivation '/nix/store/nrxks892dp91m0g748c3i62ln45p1i00-source.drv':
         specified: sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
            got:    sha256-jCzF0REocpnP56LfY42zlGTXyKyz4GPovDshhrh4jyo=
error: 1 dependencies of derivation '/nix/store/2zvhfid0wxfa370f1q182j3xm5mgq4h4-mdds-2.0.2.drv' failed to build

(...update hash in editor...)

$ grep sha256 pkgs/development/libraries/mdds/default.nix
    sha256 = "sha256-jCzF0REocpnP56LfY42zlGTXyKyz4GPovDshhrh4jyo=";

$ nix build --no-link --option substituters https://cache.nixos.org -L .#mdds
warning: Git tree '/home/endgame/code/nixpkgs' is dirty
mdds> unpacking sources
mdds> unpacking source archive /nix/store/iilsw18kwyndw7rrj6qmq0ppk723sjmg-source
mdds> source root is source
mdds> patching sources
mdds> configuring
mdds> no configure script, doing nothing
mdds> building
mdds> no Makefile, doing nothing
mdds> installing
mdds> install flags: SHELL=/nix/store/dsd5gz46hdbdk2rfdimqddhq6m8m8fqs-bash-5.1-p16/bin/bash install
mdds> make: *** No rule to make target 'install'.  Stop.
error: builder for '/nix/store/hdrknn3slq8pjyzvavyxfhv2hfk9jiif-mdds-2.0.2.drv' failed with exit code 2;
       last 10 log lines:
       > unpacking source archive /nix/store/iilsw18kwyndw7rrj6qmq0ppk723sjmg-source
       > source root is source
       > patching sources
       > configuring
       > no configure script, doing nothing
       > building
       > no Makefile, doing nothing
       > installing
       > install flags: SHELL=/nix/store/dsd5gz46hdbdk2rfdimqddhq6m8m8fqs-bash-5.1-p16/bin/bash install
       > make: *** No rule to make target 'install'.  Stop.
       For full logs, run 'nix log /nix/store/hdrknn3slq8pjyzvavyxfhv2hfk9jiif-mdds-2.0.2.drv'.
```

Reply:
Disregard, I had things backwards in my head.

Reply:
I was talking about your `fetchurl` call that used the cached tarball:

```console
$ nix store prefetch-file https://gitlab.com/mdds/mdds/-/archive/2.0.2/mdds-2.0.2.tar.bz2 --json | nix run nixpkgs#jq -- -r .hash
sha256-ClttiGc3RmDpQWwztMtH4Vr1zo50KRyv+KT1aYNOESw=
```